File: backend/src/029-concurrencia/hilos/lambdas/lamb_thread.cpp
#include <iostream>
#include <thread>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    // Vector de workers
    vector<thread> workers;

    for (int i=1; i<6; i++) {
        // Los hilos pueden tomar una función lambda
        workers.push_back(thread( [i]() {
            cout << "Dentro del hilo número " << i << "\n";
        }));
    }

    // Esperamos a finalizar todos los hilos
    for_each(workers.begin(), workers.end(), [](thread &t) {
        t.join();
    });

    // Los hilos se ejecutan de forma desordenada

    return 0;
}

File: backend/src/029-concurrencia/hilos/lambdas/lamb_thread.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import threading

workers = []

for i in range(1, 6):
    workers.append(
        threading.Thread(target=lambda arg: print("Dentro del hilo número %d" % arg),
                         args=[i])
    )

# Los hilos se ejecutan de forma ordenada
for t in workers:
    t.start()
